tests/app/submitter/test_converter.py

```diff
@@ -131,6 +134,40 @@ def test_convert_answers(self):
             self.assertEqual(answer_object['data']['001'], '2016-01-01')
             self.assertEqual(answer_object['data']['002'], '2016-03-30')
 
+    def test_convert_answers_to_data_with_key_error(self):
+        with self.application.test_request_context():
+            user_answer = [create_answer('ABC', '2016-01-01', group_id='group-1', block_id='block-1')]
+
+            questionnaire = {
+                "survey_id": "021",
+                "data_version": "0.0.1",
+                "groups": [
+                    {
+                        "id": "group-1",
+                        "blocks": [
+                            {
+                                "id": "block-1",
+                                "questions": [{
+                                    "id": 'question-1',
+                                    "answers": [
+                                        {
+                                            "id": "LMN",
+                                            "type": "TextField",
+                                            "q_code": "001"
+                                        },
+                                    ]
+                                }]
+                            }
+                        ]
+                    }
+                ]
+            }
+
+            routing_path = [Location(group_id='group-1', group_instance=0, block_id='block-1')]
+            with patch('app.submitter.converter.logger') as patched_logger:
+                convert_answers(AnswerStore(user_answer), questionnaire, routing_path)
+                self.assertEqual(patched_logger.info.call_count, 1)
```

Comment:
Change to `...patched_logger.error...` once changes to logging in [converter](https://github.com/ONSdigital/eq-survey-runner/blob/6c2d5cbaed01fd3fbc7407f806b4c370b1fd1289/app/submitter/converter.py#L95) have been applied.